tests/unit/wc-payment-api/test-class-wc-payments-api-client.php

```diff
@@ -763,6 +763,131 @@ public function test_add_tos_agreement() {
 		$this->assertEquals( [ 'result' => 'success' ], $result );
 	}
 
+	/**
+	 * @dataProvider data_request_with_level3_data
+	 */
+	public function test_request_with_level3_data( $input_args, $expected_level3_args ) {
+		$this->mock_http_client
+			->expects( $this->once() )
+			->method( 'remote_request' )
+			->with(
+				$this->anything(),
+				$this->callback(
+					function( $request_args_json ) use ( $expected_level3_args ) {
+						$request_args = json_decode( $request_args_json, true );
+
+						$this->assertSame( $expected_level3_args, $request_args['level3'] );
+
+						return true;
+					}
+				)
+			)
+			->willReturn(
+				[
+					'body'     => wp_json_encode( [ 'result' => 'success' ] ),
+					'response' => [
+						'code'    => 200,
+						'message' => 'OK',
+					],
+				]
+			);
+
+		PHPUnit_Utils::call_method(
+			$this->payments_api_client,
+			'request_with_level3_data',
+			[ $input_args, 'intentions', 'POST' ]
+		);
+	}
+
+	/**
+	 * Data provider for test_request_with_level3_data
+	 */
+	public function data_request_with_level3_data() {
+		$base_input = [
```

Comment:
I don't think you need `$base_input`. Try without it, test will still pass correctly because `request_with_level3_data()` only cares about `level3`.

Reply:
Thanks Shendy! I don't think my coffee had kicked in by that point 😄 I have removed the redundant `$base_input`.